### DISCUSSION TITLE
user_dma出現deadlock

### DISCUSSION THREAD

**Discussion by DonghanTsai** [bol-edu/HLS-SOC-Discussions#210] [General]
![螢幕擷取畫面 2024-05-01 230244](https://github.com/bol-edu/HLS-SOC-Discussions/assets/71302677/02f8283c-e9e0-4bc7-ac68-a70f6d5c3fee)
如圖，在執行過程中跑出了以下的error，看起來是user_dma中的streamtoparallelwithburst與getinstream之間出現deadlock導致的，deadlock其中一邊的原因是incount為空，另外一邊不太清楚發生什麼事，但我沒有動裡面除了BUF_LEN以外的東西，不知道有沒有其他可能造成此deadlock的原因或解決方式

**Comment by michael61112** [comment]
你是把BUF_LEN調小嗎? incount 若為空會無法累加而跳不出while loop喔

**Reply by DonghanTsai** [reply]
我調小到600 (FIR的輸入資料數)，因為我想說應該不用像原本的圖片開那麼大，所以會是這邊出問題嗎

**Reply by nthuyouwei** [reply]
@DonghanTsai  我可以詢問您後來怎麼解決的嗎?目前也遇到這個問題

**Reply by DonghanTsai** [reply]
按照這個同學的方法，有成功了
https://github.com/bol-edu/HLS-SOC-Discussions/discussions/217#discussioncomment-9357134
